src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Competitions/CompetitionUpdate.razor

```diff
@@ -32,12 +32,13 @@
 
     protected override async Task OnInitializedAsync()
     {
-        _competition = _navigator.ConsumeParameter<Competition>();
-        await _competitionWithChildren.Initialize(_competition.Id);
+        var competition = _navigator.ConsumeParameter<Competition>();
+        _competition = await _competitionWithChildren.Initialize(competition.Id);
```

Comment:
I don't understand. what's the difference between having `_competition` be assigned a reference from `ConsumeParameter` or from `Initialize`?

Reply:
Upon debugging I have found that the data loss behavior arises because the object returned by the ConsumeParameter at CompetitionUpdate isn't related to its children. I. e the Contestants and Phases fields are empty. When this state is persisted the data loss occurs.

However, with the object returned by the Initialize method at the INotParentBehind the fields are populated as they have been stored in-memory. 

Why the INavigator works as expected at Event but doesn't at Competition is also unclear to me.